### PR TITLE
Implement new JSON Jinja filters

### PR DESCRIPTION
**Pull request by nmaludy**
This is the corresponding PR to
StackStorm/st2#3766 that adds documentation on the new filters.

### CODE REVIEW

File: docs/source/reference/jinja.rst [hunk 1]
```diff
@@ -12,16 +12,29 @@ with Jinja in the context of |st2|. Please refer to the `Jinja docs
 Referencing Datastore Keys in Jinja
 ------------------------------------
 
-You can use ``{{st2kv.system.foo}}`` to access key ``foo`` from datastore. Note that until
-v2.1, the expression to access key ``foo`` from datastore used to be ``{{system.foo}}``
+You can use ``{{ st2kv.system.foo }}`` to access key ``foo`` from datastore. Note that until
+v2.1, the expression to access key ``foo`` from datastore used to be ``{{ system.foo }}``
 but is now deprecated, and the leading ``st2kv.`` namespace is required.
 
+Currently, all data in the datastore is represented as strings. Toto represent
```
Comment (armab) [inline docs/source/reference/jinja.rst]: `Toto` -> `To` typo

File: docs/source/reference/jinja.rst [hunk 2]
```diff
@@ -77,7 +107,28 @@ Adds escape characters to JSON strings.
...
+If you would like to test out your JSONPath queries prior to utilizing this filter
+an online evaluator can be found `here <http://jsonpath.com/>`.
+
+.. code-block:: bash
+
+    # input  = {'people': [{'first': 'James', 'last': 'd'},
+    #                      {'first': 'Jacob', 'last': 'e'},
+    #                      {'first': 'Jayden', 'last': 'f'}]}
+    # result = ['James', 'Jacob', 'Jayden']
+    {{ input | jmespath_query('people[*].first') }}
```
Comment (armab) [inline docs/source/reference/jinja.rst]: I think for such complex filter we need several more examples to show people how powerful it is.

Comment (nmaludy) [reply docs/source/reference/jinja.rst]: Good idea! Added more examples.

File: docs/source/reference/jinja.rst (hunk 1 above)
Comment (nmaludy) [reply docs/source/reference/jinja.rst]: Fixed, thanks!

File: docs/source/reference/jinja.rst [hunk 3]
```diff
@@ -77,7 +107,76 @@ Adds escape characters to JSON strings.
...
+    #
+    # input  = {'animals': ['bird', 'rabbit', 'cat', 'dog', 'cow'] }
+    # result = ['rabbit']
+    {{ input | jsonpath_query('animals[1]') }}
+
+    # Access all indexes in an array/list
+    #
+    # input  = {'animals': ['bird', 'rabbit', 'cat', 'dog', 'cow'] }
+    # result = ['bird', 'rabbit', 'cat', 'dog', 'cow']
+    {{ input | jsonpath_query('animals[*]') }}
```
Comment (Kami) [inline docs/source/reference/jinja.rst]: Nice work on the docs and examples :+1:

File: docs/source/reference/jinja.rst [hunk 4]
```diff
@@ -258,4 +357,4 @@ Drops patch version of supplied version field.
 
 .. code-block:: bash
 
-    {{version | version_strip_patch}}
+    {{ version | version_strip_patch }}
```
Comment (Kami) [inline docs/source/reference/jinja.rst]: Thanks, I'm also not a fan of non-whitespace approach :P

**Pull request merged**